plugin/gocode/list.go

```diff
@@ -125,11 +144,13 @@ func (s *suggestionList) apply() {
 	end := carets[0]
 	runes := s.ctrl.TextRunes()
 
-	go s.applier.Apply(s.editor, input.Edit{
-		At:  start,
-		Old: runes[start:end],
-		New: []rune(suggestion.Name),
-	})
+	if start < end {
+		go s.applier.Apply(s.editor, input.Edit{
+			At:  start,
+			Old: runes[start:end],
+			New: []rune(suggestion.Name),
+		})
+	}
```

Comment:
minor note: I prefer `if start > end {return}` here, to leave the main logic intact while bailing out for unexpected situations.